packages/amplify-cli/src/commands/env/checkout.js

```diff
@@ -22,12 +24,40 @@ module.exports = {
     const jsonString = JSON.stringify(localEnvInfo, null, 4);
     fs.writeFileSync(localEnvFilePath, jsonString, 'utf8');
 
-    // Initialize the environment
+
+    // Setup exeinfo
 
     context.amplify.constructExeInfo(context);
     context.exeInfo.forcePush = false;
+    context.exeInfo.isNewEnv = false;
     context.exeInfo.restoreBackend = context.parameters.options.restore;
 
+    // Setup Provider creds/info
+    const initializationTasks = [];
+    const providerPlugins = getProviderPlugins(context);
+    context.exeInfo.projectConfig.providers.forEach((provider) => {
+      const providerModule = require(providerPlugins[provider]);
+      initializationTasks.push(() => providerModule.init(
+        context,
+        allEnvs[envName][provider],
+      ));
+    });
+
+    await sequential(initializationTasks);
+
+    const onInitSuccessfulTasks = [];
+    context.exeInfo.projectConfig.providers.forEach((provider) => {
+      const providerModule = require(providerPlugins[provider]);
+      onInitSuccessfulTasks.push(() => providerModule.onInitSuccessful(
```

Comment:
`afterInit` or `postInitSuccess` would be a better name. Also we may want to run `afterInitFailure` just in case we want to give category plugins to clean things up

Reply:
I agree the name is a bit odd, but its the interface method name which the provider exposes and changing that would mean changing it in a bunch of places which is out of scope for this PR.

Reply:
Can we put a todo and create an issue for that. Long term name should be changed